### PR TITLE
lightning: disk quota

### PR DESCRIPTION
**Pull request by buchuitoudegou**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -367,6 +367,7 @@
     - [断点续传](/tidb-lightning/tidb-lightning-checkpoints.md)
     - [并行导入](/tidb-lightning/tidb-lightning-distributed-import.md)
     - [可容忍错误](/tidb-lightning/tidb-lightning-error-resolution.md)
+    - [磁盘配额](/tidb-lightning/tidb-lightning-disk-quota.md)
```
Comment (ran-huang) [inline TOC.md:370]: 鉴于内容较少，不建议新开一篇文档，最好加到 https://github.com/pingcap/docs-cn/blob/master/tidb-lightning/tidb-lightning-physical-import-mode-usage.md 里。 @sunzhaoyang 你看呢？

Comment (sunzhaoyang) [reply TOC.md:370]: 赞同，不算是个特别大的特性。

Comment (buchuitoudegou) [reply TOC.md]: 8b65bc9

**Comment by ran-huang** [comment]
我调整了一些语言，@buchuitoudegou PTAL at https://github.com/pingcap/docs-cn/pull/10540/files/8b65bc974d01400d205d484cc300833790267b59..a5d0a646a8a46be221154840ece494e2f414ce52

**Comment by buchuitoudegou** [comment]
> 我调整了一些语言，@buchuitoudegou PTAL at https://github.com/pingcap/docs-cn/pull/10540/files/8b65bc974d01400d205d484cc300833790267b59..a5d0a646a8a46be221154840ece494e2f414ce52

lgtm

**Review by gozssky — APPROVED** [review]

File: tidb-lightning/tidb-lightning-physical-import-mode-usage.md [hunk 2]
```diff
@@ -162,4 +162,27 @@ io-concurrency = 5
...
+
+TiDB Lightning 在使用物理模式导入数据时，会在本地磁盘创建大量的临时文件，用来对原始数据进行编码、排序、分割。当用户本地磁盘空间不足时，TiDB Lightning 会由于写入文件失败而报错退出。
+
+为了减少这种情况的发生，你可以为 TiDB Lightning 配置磁盘配额 (disk quota)。当磁盘配额不足时，TiDB Lightning 会暂停读取源数据以及写入临时文件的过程，优先将已经完成排序的 key-value 写入到 TiKV，删除本地临时文件后，再继续导入过程。
+
+为 TiDB Lightning 开启磁盘配额，你需要在配置文件中加入配置项：
+
+```toml
+[tikv-importer]
+# 默认为 MaxInt64，即 9223372036854775807 字节
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-physical-import-mode-usage.md:177]: 9223372036854775807 字节是多少 GB？最好写上，一般人看到这么长数字可能没概念。

File: tidb-lightning/tidb-lightning-physical-import-mode-usage.md [hunk 3]
```diff
@@ -162,4 +162,27 @@ io-concurrency = 5
...
+backend = "local"
+
+[cron]
+# 检查磁盘配额的时间间隔，默认为 60 秒。
+check-disk-quota = "30s"
+```
+
+`disk-quota` 配置项可以设置磁盘配额，默认为 MaxInt64，即 9223372036854775807 字节，这个值远远大于实际情况中可能需要的磁盘空间，相当于没开启。
+
+`check-disk-quota` 配置项是检查磁盘配额的时间间隔，默认为 60 秒。由于检查临时文件使用空间的过程需要加锁，会使所有的导入线程都暂停，如果在每次写入之前都检查一次磁盘空间的使用情况，则会大大降低写入文件的效率（相当于单线程写入）。为了维持高效的写入，磁盘配额不会在每次写入之前检查，而是每隔一段时间暂停所有线程的写入并检查当前磁盘空间的使用情况。也就是说，当 `check-disk-quota` 配置项设置为一个非常大的值时，磁盘的使用空间有可能会大大超出磁盘配额，这样的情况下，磁盘配额功能可以说是不生效的。因此，`check-disk-quota` 的值建议不要设置太大，而具体设置多少则需要由 TiDB Lightning 具体运行的环境决定，因为不同的环境下，TiDB Lightning 写入临时文件的速度是不一样的。理论上来说，写入临时文件的速度越快，`check-disk-quota` 需要设置得越小。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-physical-import-mode-usage.md:188]: “磁盘的使用空间有可能会大大超出磁盘配额”，这句话是不是写反了？似乎应该是 “磁盘配额有可能会大大超出磁盘的可用空间”

**Review by hfxsd — APPROVED** [review]
Rest LGTM

File: tidb-lightning/tidb-lightning-physical-import-mode-usage.md (hunk 3 above)
Comment (buchuitoudegou) [reply tidb-lightning/tidb-lightning-physical-import-mode-usage.md:188]: 没有写反，磁盘使用可能会超出磁盘配额

File: tidb-lightning/tidb-lightning-physical-import-mode-usage.md (hunk 2 above)
Comment (buchuitoudegou) [reply tidb-lightning/tidb-lightning-physical-import-mode-usage.md:177]: 9223372036 GB 对大多数人也没有概念😂这个数太大了

Comment (hfxsd) [reply tidb-lightning/tidb-lightning-physical-import-mode-usage.md:177]: 哈哈好的，不知道为啥搞这么大的数字。你们来定吧~

File: tidb-lightning/tidb-lightning-physical-import-mode-usage.md [hunk 4]
```diff
@@ -162,4 +162,27 @@ io-concurrency = 5
...
 
-此外，TiKV 的 [num-threads](/tikv-configuration-file.md#num-threads) 配置也可能影响性能，新集群建议设置为 CPU 核数。
\ No newline at end of file
+此外，TiKV 的 [num-threads](/tikv-configuration-file.md#num-threads) 配置也可能影响性能，新集群建议设置为 CPU 核数。
+
+## 磁盘资源配额 <span class="version-mark">从 v6.2.0 版本开始引入</span>
+
+TiDB Lightning 在使用物理模式导入数据时，会在本地磁盘创建大量的临时文件，用来对原始数据进行编码、排序、分割。当用户本地磁盘空间不足时，TiDB Lightning 会由于写入文件失败而报错退出。
+
+为了减少这种情况的发生，你可以为 TiDB Lightning 配置磁盘配额 (disk quota)。当磁盘配额不足时，TiDB Lightning 会暂停读取源数据以及写入临时文件的过程，优先将已经完成排序的 key-value 写入到 TiKV，删除本地临时文件后，再继续导入过程。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-physical-import-mode-usage.md:171]: 删除本地临时文件，需要用户手动执行、还是 Lightning 自动删除？需要在文档中写清楚。

Comment (ran-huang) [reply tidb-lightning/tidb-lightning-physical-import-mode-usage.md:171]: 我理解是 lightning
```suggestion
为了减少这种情况的发生，你可以为 TiDB Lightning 配置磁盘配额 (disk quota)。当磁盘配额不足时，TiDB Lightning 会暂停读取源数据以及写入临时文件的过程，优先将已经完成排序的 key-value 写入到 TiKV，TiDB Lightning 删除本地临时文件后，再继续导入过程。
```

**Pull request merged**